presto-main/src/test/java/com/facebook/presto/sql/planner/TestPlanMatchingFramework.java

```diff
@@ -201,26 +201,31 @@ public void testReferenceNonexistentAlias()
     }
 
     /*
+     * Inverts an assertion so that fails pass and passes fail.
+     *
      * There are so many ways for matches to fail that this is not likely to be generally useful.
      * Pending better diagnostics, please leave this here, and restrict its use to simple queries
      * that have few ways to not match a pattern, and functionality that is well-tested with
      * positive tests.
      */
-    private void assertFails(Runnable runnable)
+    private void assertPlanFails(Runnable runnable)
```

Comment:
I was going to comment on the function name, since this is looking for plan mismatches and not planning failures, but actually I feel this whole function isn't necessary. PlanAssert already has assertPlanDoesNotMatch, which checks the match result directly and prints a nice error message. I think the tests using this should call to that directly instead of calling a function that calls assertPlan() and then inverting it.

Reply:
Makes sense. It would be nicer to reuse more code and remove this completely. That this exists at all is probably just a relic of code written over multiple years by different devs who didn't always know exactly what already existed where. (I certainly don't.) The existing code is a little convoluted so it might take a minute to rejigger the tests to use PlanAssert.